src/nccl_ofi_net.c

```diff
@@ -158,85 +158,204 @@ void nccl_net_ofi_free_info_list(struct fi_info *info_list)
 }
 
 /*
- * @brief	Allocate a memory registration key
+ * @brief	Initialize pool of IDs
  *
- * Extract an available key from the key pool, mark the key as
- * unavailable in the key pool, and return extracted key. Noop in case
- * no key was available.
+ * Initializes a nccl_ofi_id_pool_t object, marking all IDs as available.
  *
- * This operation is locked by the key pool's internal lock.
+ * @param	id_pool
+ *		The ID pool
+ * @param	size
+ *		Size of the id pool (number of IDs)
+ * @return	0 on success
+ *		non-zero on error
+ */
+int nccl_ofi_id_pool_init(nccl_ofi_id_pool_t *id_pool, size_t size)
+{
```

Comment:
I don't know if we do this for other utils in the code, but a caller into this util has to allocate storage for id_pool and init the util all the time, so you might as well allocate it here and return a handle (and free it in _fini).

Reply:
Storage for id pool doesn't have to be allocated in the heap, but it could be a stack allocated variable. 
Another case to consider is what we have with the device struct. The id pool is part of that struct and storage is allocated when the entire device is allocated, so in that case the init doesn't have to allocate the pool. 
We could force that the ID pool is only used through pointers, but it seems an unnecessary restriction to me.